open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory

## Description of problem

My dokku dokku logs were flooded with the message in the issue title, appearing several times per second:

```
2021/05/07 17:41:35 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:35 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:35 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:35 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:36 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:36 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
2021/05/07 17:41:36 open /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: no such file or directory
```

The message didn't seem to have any effect on the actual operation. I assume that this was due to an upgrade gone wrong. It appeared on an instance running `dokku 0.24.7` on CentOS. 

The output of `dokku plugin:list` mirrored this:

```
  00_dokku-standard    0.24.7 enabled    dokku core standard plugin
  20_events            0.24.7 enabled    dokku core events logging plugin
  app-json             0.24.7 enabled    dokku core app-json plugin
  apps                 0.24.7 enabled    dokku core apps plugin
  builder              0.24.7 enabled    dokku core builder plugin
cat: /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: No such file or directory
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000088030, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc0000a4280, 0xc000088000, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:201 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000093e48, 0xc000093e78, 0x4, 0x4, 0x631220, 0x1, 0xc0000a8180, 0x18)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:83 +0x19c
github.com/progrium/go-basher.Application(0xc000093e48, 0xc000093e78, 0x4, 0x4, 0x631220, 0x665a01)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:64 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
cat: /var/lib/dokku/plugins/available/builder-cnb/plugin.toml: No such file or directory
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
main.TomlGet(0xc000088030, 0x2, 0x2)
	github.com/dokku/plugn/plugn.go:30 +0x2cd
github.com/progrium/go-basher.(*Context).HandleFuncs(0xc0000a4280, 0xc000088000, 0x5, 0x5, 0x0)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:201 +0x26d
github.com/progrium/go-basher.ApplicationWithPath(0xc000093e48, 0xc000093e78, 0x4, 0x4, 0x631220, 0x1, 0xc0000a8180, 0x18)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:83 +0x19c
github.com/progrium/go-basher.Application(0xc000093e48, 0xc000093e78, 0x4, 0x4, 0x631220, 0x665a01)
	/go/pkg/mod/github.com/progrium/go-basher@v0.0.0-20210110090515-9ffba79a02ab/basher.go:64 +0x1ce
main.main()
	github.com/dokku/plugn/plugn.go:102 +0x396
  builder-cnb                enabled
  builder-dockerfile   0.24.7 enabled    dokku core builder-dockerfile plugin
  builder-herokuish    0.24.7 enabled    dokku core builder-herokuish plugin
  builder-pack         0.24.7 enabled    dokku core builder-pack plugin
  buildpacks           0.24.7 enabled    dokku core buildpacks plugin
  certs                0.24.7 enabled    dokku core certificate management plugin
  checks               0.24.7 enabled    dokku core checks plugin
  common               0.24.7 enabled    dokku core common plugin
  config               0.24.7 enabled    dokku core config plugin
  cron                 0.24.7 enabled    dokku core cron plugin
  docker-options       0.24.7 enabled    dokku core docker-options plugin
  domains              0.24.7 enabled    dokku core domains plugin
  enter                0.24.7 enabled    dokku core enter plugin
  git                  0.24.7 enabled    dokku core git plugin
  letsencrypt          0.9.4 enabled    Automated installation of let's encrypt TLS certificates
  logs                 0.24.7 enabled    dokku core logs plugin
  mysql                1.9.1 enabled    dokku mysql service plugin
  network              0.24.7 enabled    dokku core network plugin
  nginx-vhosts         0.24.7 enabled    dokku core nginx-vhosts plugin
  plugin               0.24.7 enabled    dokku core plugin plugin
  proxy                0.24.7 enabled    dokku core proxy plugin
  ps                   0.24.7 enabled    dokku core ps plugin
  repo                 0.24.7 enabled    dokku core repo plugin
  resource             0.24.7 enabled    dokku core resource plugin
  scheduler-docker-local 0.24.7 enabled    dokku core scheduler-docker-local plugin
  shell                0.24.7 enabled    dokku core shell plugin
  ssh-keys             0.24.7 enabled    dokku core ssh-keys plugin
  storage              0.24.7 enabled    dokku core storage plugin
  tags                 0.24.7 enabled    dokku core tags plugin
  tar                  0.24.7 enabled    dokku core tar plugin
  trace                0.24.7 enabled    dokku core trace plugin
```

After asking for advice in the IRC channel, I removed all remaints of the builder-cnb via `sudo rm -rf /var/lib/dokku/plugins/available/builder-cnb /var/lib/dokku/plugins/enabled/builder-cnb /var/lib/dokku/core-plugins/available/builder-cnb /var/lib/dokku/core-plugins/enabled/builder-cnb`, which seems to have solved the problem for now.

--- a/debian/postinst
+++ b/debian/postinst
@@ -43,6 +43,20 @@ setup-storage() {
 }
 
 setup-plugins() {
+  echo "Deleting invalid plugins"
+  if [ -f "${DOKKU_LIB_ROOT}/core-plugins/available/" ]; then
+    find "${DOKKU_LIB_ROOT}/core-plugins/available/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/core-plugins/enabled/" ]; then
+    find "${DOKKU_LIB_ROOT}/core-plugins/enabled/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/plugins/available/" ]; then
+    find "${DOKKU_LIB_ROOT}/plugins/available/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/plugins/enabled/" ]; then
+    find "${DOKKU_LIB_ROOT}/plugins/enabled/" -type d -empty -delete
+  fi
+
   echo "Setting up plugin directories"
   # should be replaced by `plugn init`
   mkdir -p "${DOKKU_LIB_ROOT}/core-plugins/available" "${DOKKU_LIB_ROOT}/plugins/available"

--- a/rpm/dokku.postinst
+++ b/rpm/dokku.postinst
@@ -18,6 +18,20 @@ setup-storage() {
 }
 
 setup-plugins() {
+  echo "Deleting invalid plugins"
+  if [ -f "${DOKKU_LIB_ROOT}/core-plugins/available/" ]; then
+    find "${DOKKU_LIB_ROOT}/core-plugins/available/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/core-plugins/enabled/" ]; then
+    find "${DOKKU_LIB_ROOT}/core-plugins/enabled/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/plugins/available/" ]; then
+    find "${DOKKU_LIB_ROOT}/plugins/available/" -type d -empty -delete
+  fi
+  if [ -f "${DOKKU_LIB_ROOT}/plugins/enabled/" ]; then
+    find "${DOKKU_LIB_ROOT}/plugins/enabled/" -type d -empty -delete
+  fi
+
   echo "Setting up plugin directories"
   # should be replaced by `plugn init`
   mkdir -p "${DOKKU_LIB_ROOT}/core-plugins/available" "${DOKKU_LIB_ROOT}/plugins/available"

